### PR TITLE
[pradyuprasad] ip

### PR DESCRIPTION
**Pull request by pradyuprasad**
# 🌟 Introducing Orion: Your Personal Task Manager

Orion is a command-line-based task management application designed to help you stay organized and productive. With Orion, you can manage various tasks such as todos, deadlines, and events, all from the comfort of your terminal.

## Features

- **Task Management**:
  - Create tasks with specific descriptions and track them easily.
  - Organize your tasks into three categories: `TODO`, `DEADLINE`, and `EVENT`.

- **File Storage**:
  - Persist your tasks between sessions using a simple CSV-based storage system.
  - Automatically loads your tasks when you start the application.

- **Command Interface**:
  - Use natural language commands like `todo`, `deadline`, and `event` to interact with the app.
  - Mark tasks as completed or delete them when they are no longer needed.

## Usage Example

Here's how you can add a new deadline task in Orion:

```plaintext
// Command to add a deadline task
deadline Submit report /by 15/09/2024 2359
```

## Advanced Parsing Feature: Event Command Validation

Orion's Parser class includes advanced features for validating complex commands, such as adding an event with specific start and end times. Below is an example of how the `validateEventCommand` method ensures that the event command is correctly formatted:

```java
public class Example {
    public static void main(String[] args) {
        Parser parser = new Parser();

        // Example command input
        String[] commandParts = {"event", "Team Meeting /from 20/09/2024 1400 /to 20/09/2024 1600"};

        try {
            // Validate the event command
            EventDetails eventDetails = parser.validateEventCommand(commandParts);
            System.out.println("Valid command. Event: " + eventDetails.getDescription() +
                               " from " + eventDetails.getFrom() + " to " + eventDetails.getTo());
        } catch (InvalidEventException e) {
            System.err.println("Invalid event command: " + e.getMessage());
        }
    }
}
```

## What's Next?

1. **Enhance User Interface**:
   - Improve command feedback with more detailed messages.
   - Add support for more complex date formats.

2. **Expand Functionality**:
   - Implement recurring tasks to automate repetitive scheduling.
   - Introduce task prioritization to help users focus on the most important tasks.

3. **Increase Test Coverage**:
   - Ensure all edge cases are handled with appropriate unit tests.
   - Achieve 100% code coverage across all classes.

## Development Status

- [x] **Implement basic task management features**
- [x] **Add CSV storage support**
- [ ] *Implement recurring tasks*
- [ ] *Improve user interface and feedback*

## Additional Information

> "Orion is more than just a task manager; it's your daily productivity companion." – The Orion Team

For more information, visit the [Orion Github page](https://github.com/pradyuprasad/ip).

### CODE REVIEW

File: src/main/java/orion/storage/Storage.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] parts = line.split(",");
+                int taskId = Integer.parseInt(parts[0]);
+                String type = parts[1];
+                String description = parts[2];
+                boolean completed = Boolean.parseBoolean(parts[3]);
+
+                Task task = null;
+                switch (type) {
```
Comment (kaajinn) [inline src/main/java/orion/storage/Storage.java:102]: Should the indentation for the 'case' clauses be removed?

File: src/main/java/orion/Orion.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+     */
+    public static void main(String[] args) {
+        ui.showWelcome();
+
+        while (true) {
+            String input = ui.readUserInput();
+            String[] parts = input.split(" ", 2);
+            Command command = Command.fromString(parts[0]);
+
+            switch (command) {
```
Comment (kaajinn) [inline src/main/java/orion/Orion.java:51]: Should the indentation for the 'case' clauses be removed?

File: src/main/java/orion/task/Task.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * This is an abstract class that provides the common properties and methods
+ * for all types of tasks, such as TODOs, DEADLINEs, and EVENTs. Specific task
+ * types will extend this class and implement the {@code getTypeIcon} method
+ * to return their respective type icons.
+ * </p>
+ */
+public abstract class Task {
+    private final int taskID;
+    private final String description;
+    private boolean completed;
```
Comment (kaajinn) [inline src/main/java/orion/task/Task.java:16]: Should this be changed to 'isComplete' or something similar to more closely adhere to the coding standard?

File: src/main/java/orion/commands/Command.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package orion.commands;
+
+/**
+ * The Command enum represents the various commands that the application can
+ * recognize and execute.
+ * Each enum constant corresponds to a specific command that the user can input.
+ */
+public enum Command {
```
Comment (kaajinn) [inline src/main/java/orion/commands/Command.java:8]: I like how clean this is 👍, very organized way of managing commands!

**Review by kaajinn — COMMENTED** [review]
LGTM! Only very minor nits, otherwise a fantastic job!

File: src/main/java/orion/Orion.java [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                case DELETE:
+                    handleDelete(parts);
+                    break;
+                case FIND:
+                    handleFind(parts);
+                    break;
+                case UNKNOWN:
+                default:
+                    ui.showError("Unknown command: " + parts[0]);
+            }
```
Comment (kaajinn) [inline src/main/java/orion/Orion.java]: Should the indentation for the 'case' clauses be removed?

File: src/main/java/orion/commands/Command.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    MARK,
+    UNMARK,
+    TODO,
+    DEADLINE,
+    EVENT,
+    DELETE,
+    BYE,
+    FIND,
+    UNKNOWN;
+
```
Comment (kaajinn) [inline src/main/java/orion/commands/Command.java:19]: I like how clean this is 👍, very organized way of managing commands!

File: src/main/java/orion/storage/Storage.java [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+                        break;
+                    case "EVENT":
+                        String[] eventTimes = parts[4].split("\\|");
+                        LocalDateTime from = LocalDateTime.parse(eventTimes[0]);
+                        LocalDateTime to = LocalDateTime.parse(eventTimes[1]);
+                        task = new Event(taskId, description, from, to);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unknown task type: " + type);
+                }
```
Comment (kaajinn) [inline src/main/java/orion/storage/Storage.java]: Should the indentation for the 'case' clauses be removed?

**Review by kaajinn — COMMENTED** [review]
Fixed the formatting for some of the previous comments I left for easier reading. My bad!

**Comment by clarissatjx** [comment]
Naming conventions in general were great, no major changes to be made (with regards to second PR review coding standards). Your code is really neat and easy to read, keep it up :)
